### PR TITLE
ARM64 - Do not emit possible DIV_BY_ZERO/OVERFLOW exception blocks for non-zero/non-(negative one) constants

### PR DESCRIPTION
**Pull request by TIHan**
**Description**

I noticed that we do not have to emit DIV_BY_ZERO/OVERFLOW exceptions if we know the second operand is a non-zero/non-(negative) integer constant respectively.

**Acceptance Criteria**

- [x] CI passes

### CODE REVIEW

File: src/coreclr/jit/morph.cpp [hunk 1]
```diff
@@ -11701,7 +11701,12 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             {
                 // Codegen for this instruction needs to be able to throw two exceptions:
                 fgAddCodeRef(compCurBB, bbThrowIndex(compCurBB), SCK_OVERFLOW);
```
Comment (jkotas) [inline src/coreclr/jit/morph.cpp:11703]: Add similar condition for the overflow exception as well? It should be needed only if the second operand is -1.

Comment (TIHan) [reply src/coreclr/jit/morph.cpp:11703]: Sure, makes sense.

**Comment by TIHan** [comment]
@dotnet/jit-contrib PR is ready. I don't think there is anything else to be done here.

**Review by BruceForstall — APPROVED** [review]
Does the -1 case show up in diffs? If not, can you add a test case?

**Comment by TIHan** [comment]
> Does the -1 case show up in diffs? If not, can you add a test case?

If the constant was -1, we shouldn't see a diff as the overflow block will still be emitted.

What we will see is the regression diffs from this: https://github.com/dotnet/runtime/pull/68885#issuecomment-1119173082
Will be much less and the overflow check won't be emitted.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Pull request merged**